presto-orc/src/main/java/com/facebook/presto/orc/reader/StreamReader.java

```diff
@@ -24,6 +25,12 @@
     void readBatch(Object vector)
             throws IOException;
 
+    default void readBatch(Type type, Object vector)
```

Comment:
Having a second `readBatch` seems suboptimal. i think you can avoid this and clean up some other code by:
1) change `SingleObjectVector` to `SingleBlockBuilderVector` which has a `BlockBuilder`
2) construct that `BlockBuilder` in `LazyStructuralBlockLoader.load()` where you have the `Type` object

then you can remove this method and just use the `BlockBuilder` in `BlockStreamReader`